move static channelmanager functions into their own file

Evanfeenstra opened the pull request:
- new `ln::onion_payment` file with static functions from `channelmanager.rs`.
- cleaned up `test_peel_payment_onion` and moved into `ln::onion_payment`.
- made `PendingHTLCInfo.payment_hash` public (mistake from https://github.com/lightningdevkit/rust-lightning/pull/2700).
- made `UpdateAddHTLC.onion_routing_packet` public, so that `peel_payment_onion` can be used externally.
- some better comments on newly public fields.

jkczyz commented:
I think `channelmanager_utils` is too broad. Let's use a more restrictive name lest it become another grab bag of utilities, even if only related to `ChannelManager`. Examples of other modules that we've refactored out are `inbound_payment` and `outbound_payment`. Maybe `onion_payment` in contrast to `onion_message`? Open to other ideas.

shaavan reviewed (commented):
Concept ACK

The refactoring looks fantastic on my initial review, and the additional comments are really helpful! I'll take another pass at the added code to see if there are any further improvements we can make to this pull request. Superb work!

Evanfeenstra commented on lightning/src/ln/channelmanager.rs:
```diff
@@ -118,14 +119,19 @@ pub enum PendingHTLCRouting {
 		incoming_cltv_expiry: u32,
-		/// Optional shared secret for phantom node.
+		/// Shared secret derived using a phantom node secret key. If this field is Some, the
+		/// payment was sent to a phantom node (one hop behond the current node), but can be
```
hehe thanks for catching that!

Evanfeenstra commented:
> I think `channelmanager_utils` is too broad. Let's use a more restrictive name lest it become another grab bag of utilities, even if only related to `ChannelManager`. Examples of other modules that we've refactored out are `inbound_payment` and `outbound_payment`. Maybe `onion_payment` in contrast to `onion_message`? Open to other ideas.

Good point, I renamed the module to `onion_payment`

shaavan commented on lightning/src/ln/channelmanager.rs:
```diff
@@ -118,14 +119,19 @@ pub enum PendingHTLCRouting {
 		short_channel_id: u64, // This should be NonZero<u64> eventually when we bump MSRV
 	},
 	/// An HTLC paid to an invoice we generated.
```
Small nit:

In the following line, we are saying we don't know yet if the invoice was generated by us. So adding a "supposedly" will make the comments more coherent here.

```suggestion
	/// An HTLC paid to an invoice (supposedly) generated by us.
```

shaavan commented on lightning/src/ln/onion_payment.rs:
```diff
@@ -0,0 +1,499 @@
+/// acceptance. If the payment is to be received, and the amount matches the expected amount for
+/// a given invoice, this indicates the [`msgs::UpdateAddHTLC`], once fully committed in the
+/// channel, will generate an [`Event::PaymentClaimable`].
+/// 
```
nit by linter:

```suggestion
///
```

shaavan commented on lightning/src/ln/onion_payment.rs:
```diff
@@ -0,0 +1,499 @@
+//! Utilities for channelmanager.rs
+//! 
```
nit by linter:

```suggestion
//!
```

shaavan reviewed (commented):
ACK mod nit.

This is a refactoring PR that moves static channelmanger functions to their own file.

I have used the following line command to notice the essential differences introduced by refactoring:

```bash
git diff HEAD^ --color-moved-ws=ignore-all-space --color-moved=dimmed-zebra
```

**Essential Changes in the PR:**

1. Updated comments for members of `PendingHTLCRouting` enum. The comments look clean and clear and present a better understanding of the member variables. 
2. Modified the visibility of moved code to pub(super). Since these are utilities of `channelManager` and are not meant to be used in the rest of the codebase, this visibility is the most apt one for it.
3. Removed `create_payment_onion` function. Instead, use the `create_payment_onion` function defined in `ln::onion_utils::create_payment_onion`;

The refactoring looks clean and clear, and other than the small nits, this PR is good to be merged.

Evanfeenstra commented:
ok, rebased to `main` and incorporated the nits by @shaavan

shaavan reviewed (approved):
Code review ACK

valentinewallace reviewed (approved):
Fine to save these for follow-up if we want to get the initial code move in.

valentinewallace commented:
Will wait til @TheBlueMatt ACK's the docs to merge.

valentinewallace commented:
Going to land this since it's largely a code move and we can continue iterating on the docs if necessary in #2752.
